lms/static/js/discovery/app.js

```diff
@@ -1,70 +1,72 @@
 ;(function (define) {
-
-define(['backbone', 'course_discovery_meanings'], function(Backbone, meanings) {
     'use strict';
 
-    return function (Collection, Form, ResultListView, FilterBarView, FacetsBarView, searchQuery) {
-        //facet types configuration - set default display names
-        var facetsTypes = meanings;
+    define(['backbone', 'js/discovery/collection', 'js/discovery/form', 'js/discovery/result_list_view',
```

Comment:
Or fix all the main/app ugliness and call them factories. In any case, make sure that @dsego reviews when the dust settles (I'll trust you guys to work this out!).